sys/fork.c

```diff
@@ -0,0 +1,57 @@
+#include <fork.h>
+#include <thread.h>
+#include <sync.h>
+#include <filedesc.h>
+#include <sched.h>
+#include <stdc.h>
+
+int do_fork() {
+  thread_t *td = thread_self();
+
+  /* Cannot fork non-user threads. */
+  assert(td->td_uspace);
+
+  thread_t *newtd = thread_create(td->td_name, NULL, NULL);
+
+  /* Clone the thread. Since we don't use fork-oriented thread_t layout, we copy
```

Comment:
According to POSIX.1 standard, the only thread that passes to child process is the one that called `fork()`.